### PR TITLE
Publish Binaries

### PR DESCRIPTION
**Pull request by perrymanuk**
Created an action to publish binary to the github repo for linux. Will
need to have the github token secret added to the repo's secrets in
order to work.

### CODE REVIEW

**Comment by hendrikmaus** [comment]
> Will need to have the github token secret added to the repo's secrets in order to work.

Actually that is what `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}` is for. It is the default token to perform actions in the name of the GitHub Actions runner.

File: .github/workflows/publish-binaries.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+on:
+  release:
+    types: [created]
```
Comment (hendrikmaus) [inline .github/workflows/publish-binaries.yaml:3]:
```suggestion
    types:
      - published
```

File: .github/workflows/publish-binaries.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  release:
+    name: release ${{ matrix.target }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target: [x86_64-unknown-linux-musl]
+    steps:
+      - uses: actions/checkout@master
```
Comment (hendrikmaus) [inline .github/workflows/publish-binaries.yaml:14]:
```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/publish-binaries.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      - name: Compile and release
+        uses: rust-build/rust-build.action@latest
```
Comment (hendrikmaus) [inline .github/workflows/publish-binaries.yaml:16]:
```suggestion
        uses: rust-build/rust-build.action@v1.0.0
```

**Review by hendrikmaus — COMMENTED** [review]
I'd like to pin the versions of the tools and change the event we run on.

**Comment by perrymanuk** [comment]
> > Will need to have the github token secret added to the repo's secrets in order to work.
> 
> Actually that is what `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}` is for. It is the default token to perform actions in the name of the GitHub Actions runner.

Cool, did not know that worked like this :tada:

**Review by hendrikmaus — APPROVED** [review]

**Pull request merged**